components/tor/onion_location_navigation_throttle.cc

```diff
@@ -70,13 +70,15 @@ OnionLocationNavigationThrottle::WillProcessResponse() {
   // https://gitweb.torproject.org/tor-browser-spec.git/plain/proposals/100-onion-location-header.txt
   if (headers && GetOnionLocation(headers, &onion_location) &&
       !navigation_handle()->GetURL().DomainIs("onion")) {
+    GURL url(onion_location);
+    if (!url.SchemeIsHTTPOrHTTPS())
```

Comment:
@darkdh the onion location spec also says:

   * The Onion-Location value must be a valid URL with http: or https: protocol
     and a .onion hostname.

   * The webpage defining the Onion-Location header must be served over HTTPS.

   * The webpage defining the Onion-Location header must not be an onionsite.
   
   do we check somewhere that the webpage which sets the header is HTTPS, and that the onion-location host ends in `.onion`?

Reply:
> do we check somewhere that the webpage which sets the header is HTTPS

We do not currently do that: http://fmarier.com/

Reply:
good catch, I just pushed the fix to cover that.